deblur/workflow.py

```diff
@@ -401,12 +396,101 @@ def merge_otu_tables(output_fp, all_tables):
     all_tables: list
         list of filepaths for BIOM tables to merge
     """
+    logger = logging.getLogger(__name__)
+    logger.debug('merge_otu_tables for file %d tables'
+                 ' into file %s' % (len(all_tables), output_fp))
+
     master = load_table(all_tables[0])
     for input_fp in all_tables[1:]:
         master = master.merge(load_table(input_fp))
     write_biom_table(master, output_fp)
 
 
+def get_files_for_table(input_dir,
+                        file_end='.fasta.trim.derep.no_artifacts'
+                        '.msa.deblur.no_chimeras'):
+    """Get a list of files to add to the output table
+
+    Parameters:
+    -----------
+    input_dir : string
+        name of the directory containing the deblurred fasta files
+    file_end : string
+        the ending of all the fasta files to be added to the table
+        (default '.trim.derep.no_artifacts.msa.deblur.no_chimeras')
+
+    Returns
+    -------
+    names : list of tuples of (string,string)
+        list of tuples of:
+            name of fasta files to be added to the biom table
+            sampleid (file names without the file_end and path)
+    """
+    logger = logging.getLogger(__name__)
+    logger.debug('get_files_for_table input dir %s, '
+                 'file-ending %s' % (input_dir, file_end))
+
+    names = []
+    for cfile in listdir(input_dir):
+        cname = join(input_dir, cfile)
+        if not isfile(cname):
+            continue
+        if len(cfile) < len(file_end):
+            continue
+        if cfile[-len(file_end):] == file_end:
+            sampleid = cfile[:-len(file_end)]
+            names.append((cname, sampleid))
+    logger.debug('found %d files' % len(names))
+    return names
+
+
+def create_otu_table(output_fp, deblurred_list):
```

Comment:
i think it didn't show because this file failed the flake8... fixed and
re-pushed...

On Mon, Jun 20, 2016 at 5:22 PM, Jose Navas <notifications@github.com>
wrote:

> In deblur/workflow.py
> <https://github.com/biocore/deblur/pull/34#discussion_r67790290>:
>
> > +
> > +    names = []
> > +    for cfile in listdir(input_dir):
> > +        cname = join(input_dir, cfile)
> > +        if not isfile(cname):
> > +            continue
> > +        if len(cfile) < len(file_end):
> > +            continue
> > +        if cfile[-len(file_end):] == file_end:
> > +            sampleid = cfile[:-len(file_end)]
> > +            names.append((cname, sampleid))
> > +    logger.debug('found %d files' % len(names))
> > +    return names
> > +
> > +
> > +def create_otu_table(output_fp, deblurred_list):
>
> @amnona <https://github.com/amnona> this comment has not been addressed
> (it applies to the function create_otu_table
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/biocore/deblur/pull/34/files/3571ea3b02687547816e6f8122ff2ca3628505d1#r67790290>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFkA8tS9QQZDMFrJWz3HWjx7pMjNdgfIks5qNy6xgaJpZM4IxpPJ>
> .
>